src/lib/image/image.c

```diff
@@ -222,6 +223,50 @@ void singularity_limit_container_owners(struct image_object *image) {
     }
 }
 
+void singularity_limit_container_groups(struct image_object *image) {
+    const char *limit_container_groups = singularity_config_get_value(LIMIT_CONTAINER_GROUPS);
+
+    if ( strcmp(limit_container_groups, "NULL") != 0 ) {
+        struct stat image_stat;
+        char *group_token = NULL;
+        char *current = strtok_r(strdup(limit_container_groups), ",", &group_token);
```

Comment:
This may be a very naive question from a dummy who doesn't know c very well, but why the `strdup` here?  Doesn't it work just the same without the `strdup`?

Reply:
ping @cclerget.  Can you explain this to me when you get a chance?

Reply:
Thank you for the clarification!